include/margo-monitoring.h

```diff
@@ -0,0 +1,397 @@
+/**
+ * @file margo-monitoring.h
+ *
+ * (C) 2022 The University of Chicago
+ *
+ * See COPYRIGHT in top-level directory.
+ */
+#ifndef __MARGO_MONITORING_H
+#define __MARGO_MONITORING_H
+
+#include <abt.h>
+#include <mercury.h>
+#include <mercury_types.h>
+#include <mercury_bulk.h>
+
```

Comment:
I'm not  sure where to put this remark, but it seems like it would be helpful somewhere within the source tree to have a brief, high-level, conceptual description of the monitoring functionality.  Could be a .md file or maybe even a block comment here.  I know it will have rtd docs later that will be thorough, so it's just a conceptual overview needed here.

Reply:
Yes, I am planning to do that.

Reply:
I pushed a new commit with a lot of comments in the header.